caffe2/operators/reduce_ops.cc

```diff
@@ -1,7 +1,127 @@
 #include "caffe2/operators/reduce_ops.h"
 
+#include <algorithm>
+#include <functional>
+#include <vector>
+
 namespace caffe2 {
 
+namespace {
+
+void IncreaseIndexInDims(const int n, const int* dims, int* index) {
```

Comment:
Nit: it seems we have the same functions in math_cpu.cc. Shall we merge them and reuse?